cl/citations/tests.py

```diff
@@ -1727,3 +1731,316 @@ def test_get_graph_component(self):
+        )
+
+        # Here opinion cluster 2 has the citation 56 F.2d 9, but the
+        # HTML with citations contains star pagination for pages 9 and 10.
+        # This tests if we can find opinion cluster 2 with page 9 and 10
+        r = await self.async_client.post(
+            reverse("citation-lookup-list", kwargs={"version": "v3"}),
+            {"reporter": "f2d", "volume": "56", "page": "10"},
+        )
+
+        self.assertEqual(r.status_code, HTTPStatus.OK)
+        data = json.loads(r.content)
+        self.assertEqual(len(data), 1)
+
+        first_citation = data[0]
+        self.assertEqual(first_citation["citation"], "56 f2d 10")
+        self.assertEqual(first_citation["status"], HTTPStatus.OK)
+
+        clusters = first_citation["clusters"]
+        self.assertEqual(len(clusters), 1)
+        self.assertEqual(
+            clusters[0]["absolute_url"],
+            self.opinion_cluster_2.get_absolute_url(),
+        )
+
+    async def test_can_handle_reporter_typos(self):
+        r = await self.async_client.post(
+            reverse("citation-lookup-list", kwargs={"version": "v3"}),
+            {"reporter": "F2d", "volume": "56", "page": "9"},
+        )
+        self.assertEqual(r.status_code, HTTPStatus.OK)
+        data = json.loads(r.content)
+        self.assertEqual(len(data), 1)
+
+        first_citation = data[0]
+        self.assertEqual(first_citation["citation"], "56 F2d 9")
+        self.assertEqual(first_citation["status"], HTTPStatus.OK)
+
+        clusters = first_citation["clusters"]
+        self.assertEqual(len(clusters), 1)
+        self.assertEqual(
+            clusters[0]["absolute_url"],
+            self.opinion_cluster_2.get_absolute_url(),
+        )
+
+        # Introduce a space into the reporter
+        r = await self.async_client.post(
+            reverse("citation-lookup-list", kwargs={"version": "v3"}),
+            {"reporter": "f 2d", "volume": "56", "page": "9"},
+        )
+        self.assertEqual(r.status_code, HTTPStatus.OK)
+        data = json.loads(r.content)
+        self.assertEqual(len(data), 1)
+
+        first_citation = data[0]
+        self.assertEqual(first_citation["citation"], "56 f 2d 9")
+        self.assertEqual(first_citation["status"], HTTPStatus.OK)
+
+        clusters = first_citation["clusters"]
+        self.assertEqual(len(clusters), 1)
+        self.assertEqual(
+            clusters[0]["absolute_url"],
+            self.opinion_cluster_2.get_absolute_url(),
+        )
+
+    async def test_can_handle_full_citation_within_text(self) -> None:
+        """Do we get redirected to the correct URL when we pass in a full
+        citation?"""
+        text_citation = (
+            "Reference to Lissner v. Saad, 56 F.2d 9 11 (1st Cir. 2015)"
+        )
+        r = await self.async_client.post(
+            reverse("citation-lookup-list", kwargs={"version": "v3"}),
+            {"text_citation": text_citation},
```

Comment:
I'm sorry, but I don't think I like the POST key to be `text_citation`. A couple reasons:

1. It's wordy. Seems like something simpler is possible.
2. We can have more than one citation, but it implies it's singular. 

Could we just have the key be `text` instead?

Reply:
> Could we just have the key be text instead?

sure! I'll tweak the serializer.